src/ai/backend/manager/models/image.py

```diff
@@ -142,44 +143,74 @@ async def rescan_images(
     *,
     reporter: ProgressReporter | None = None,
 ) -> None:
+    def filter_registry_dict(
+        registry_dict: dict[str, ContainerRegistryRow],
+        condition: Callable[[str, ContainerRegistryRow], bool],
+    ) -> dict[str, ContainerRegistryRow]:
+        return {
+            registry_key: registry_row
+            for registry_key, registry_row in registry_dict.items()
+            if condition(registry_key, registry_row)
+        }
+
```

Comment:
In terms of readability, I think it’s better to define the function outside rather than using a closure in this case. The function is getting too long.